go/tools/builders/cgo.go

```diff
@@ -189,6 +195,34 @@ func run(args []string) error {
 	if err := cmd.Run(); err != nil {
 		return fmt.Errorf("error running cgo: %v", err)
 	}
+	// Now we fix up the generated files
+	trim := linePrefix + abs(".")
+	for _, src := range cgoOuts {
+		body, err := ioutil.ReadFile(src)
+		if err != nil {
+			return err
+		}
+		out, err := os.Create(src)
```

Comment:
Instead of overwriting the original file here, write to a new file, then defer `os.Rename` / `os.Remove` at the end for commit / rollback. Better to keep the original file intact in case of error instead of a partial file.